quesma/clickhouse/parser.go

```diff
@@ -141,9 +123,12 @@ func RemoveNonSchemaFields(m SchemaMap, t *Table) SchemaMap {
 				descendRec(col, vCasted)
 			}
 		} else {
-			delete(m, fieldName) // TODO check if it's fine? In c++ not, but here seems to work
+			toDelete = append(toDelete, fieldName)
```

Comment:
Hmm, the internet says it's OK to delete elements from map while iterating over it? (so the previous, simpler approach)

https://stackoverflow.com/questions/23229975/is-it-safe-to-remove-selected-keys-from-map-within-a-range-loop

Reply:
That's interesting, but I was basing on other post that was telling that's unsafe